crates/ruff/src/rules/flake8_todo/rules.rs

```diff
@@ -0,0 +1,384 @@
+/// ```
+///
+/// Use instead
+/// ```python
+/// # TODO(ruff): now an author is assigned
+/// ```
+#[violation]
+pub struct MissingAuthorInTodo;
+impl Violation for MissingAuthorInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing author in TODO")
+    }
+}
+
+/// ## What it does
+/// Checks that an issue link or ticket is associated with a TODO.
+///
+/// ## Why is this bad?
+/// Including an issue link near a TODO makes it easier for resolvers
+/// to get context around the issue and keeps code formatting consistent.
+///
+/// ## Example
+/// ```python
+/// # TODO: this link has no issue
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): solve this issue!
+/// # https://github.com/charliermarsh/ruff/issues/3870
+/// ```
+#[violation]
+pub struct MissingLinkInTodo;
+impl Violation for MissingLinkInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing issue link following TODO")
+    }
+}
+
+/// ## What it does
+/// Checks that a "TODO" tag is followed by a colon.
+///
+/// ## Why is this bad?
+/// Skipping colons after a "TODO" tag can create inconsistent code formatting.
+///
+/// ## Example
+/// ```python
+/// # TODO(ruff) fix this colon
+/// ```
+///
+/// Used instead:
+/// ```python
+/// # TODO(ruff): colon fixed
+/// ```
+#[violation]
+pub struct MissingColonInTodo;
+impl Violation for MissingColonInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing colon in TODO")
+    }
+}
+
+/// ## What it does
+/// Checks that a "TODO" tag has some text after it.
+///
+/// ## Why is this bad?
+/// Just putting a "TODO" tag in the code, without any context, makes it harder
+/// for the reader/future resolver to understand the issue that has to be fixed.
+///
+/// ## Example
+/// ```python
+/// # TODO(ruff)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): fix some issue
+/// ```
+#[violation]
+pub struct MissingTextInTodo;
+impl Violation for MissingTextInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Missing text in TODO")
+    }
+}
+
+/// ## What it does
+/// Checks that a "TODO" tag is properly capitalized, i.e. that the tag is uppercase.
+///
+/// ## Why is this bad?
+/// Inconsistent capitalization leads to less readable code.
+///
+/// ## Example
+/// ```python
+/// # todo(ruff): capitalize this
+/// ```
+///
+/// Use instead:
+/// ```python
+/// # TODO(ruff): this is capitalized
+/// ```
+#[violation]
+pub struct InvalidCapitalizationInTodo {
+    pub tag: String,
+}
+impl AlwaysAutofixableViolation for InvalidCapitalizationInTodo {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let InvalidCapitalizationInTodo { tag } = self;
+        format!("Invalid TODO capitalization: `{tag}` should be `TODO`")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Fix capitalization in `TODO`".to_string()
```

Comment:
Autofix titles gives suggestion on what to do, so I think this might be more helpful and inline with other titles:
```suggestion
        let InvalidCapitalizationInTodo { tag } = self;
        "Replace `{tag}` with `TODO`".to_string()
```